src/library/scala/concurrent/duration/Duration.scala

```diff
@@ -124,6 +124,8 @@ object Duration {
       Undefined
     else if (nanos > Long.MaxValue || nanos < Long.MinValue)
       throw new IllegalArgumentException("trying to construct too large duration with " + nanos + "ns")
+    else if (nanos < 0)
+      fromNanos((nanos - 0.5).toLong)
```

Comment:
I don't really have an opinion on exactly how and where to round, but I agree that `Duration("-1 d")` should be `1 Day` and not `-86399999999999 nanoseconds`. Whether everything gets rounded down, or gets scala.math.rounded, or something else, I have no opinion on.